### PR TITLE
(DOCSP-26228): Update tutorial to not delete all app data

### PR DESCRIPTION
**Pull request by mongodben**
## Pull Request Info

Update tutorial to not delete all app data. This step is no longer necessary b/c of a bug fix in Realm Flutter SDK v0.6.0+beta.

This change **must** be merged after the changes from https://github.com/mongodb-university/realm-template-apps/pull/102 have been deployed for use in the Realm CLI. (However, the changes don't need to be merged in tandem, b/c the tutorial as currently documented still works with the previous version..just the app deletion step is no longer necessary.)

### Jira

- https://jira.mongodb.org/browse/DOCSP-26228

### Staged Changes (Requires MongoDB Corp SSO)

- [Flutter Device Sync Tutorial](https://docs-atlas-staging.mongodb.com/atlas-app-services/docsworker-xlarge/DOCSP-26228/tutorial/flutter/)

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-app-services/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/tutorial/flutter.txt [hunk 1]
```diff
@@ -209,24 +209,18 @@ Modify the Application
...
+      sync errors.
 
-      At this point, you can run the application again. Log in using the account
-      you created earlier in this tutorial. You will see the one Item you
-      previously created. Add a new Item, and you will see that you can now
-      set the priority. Choose ``High`` for the priority and save the Item.
+      Then, Log in using the account you created earlier in this tutorial.
+      You will see the one Item you previously created.
+      Add a new Item, and you will see that you can now set the priority.
+      Choose ``High`` for the priority and save the Item.
```
Comment (mongodben) [inline source/tutorial/flutter.txt:221]: note: no copy changes here. just moved text across lines for formatting.

File: source/tutorial/flutter.txt [hunk 2]
```diff
@@ -209,24 +209,16 @@ Modify the Application
...
-      previously created. Add a new Item, and you will see that you can now
-      set the priority. Choose ``High`` for the priority and save the Item.
+      Before you run the application again, perform a `hot restart <https://docs.flutter.dev/development/tools/hot-reload#special-cases>`__.
+      This makes sure that the sync session restarts with the new schema and prevents
+      sync errors.
+
+      Then, Log in using the account you created earlier in this tutorial.
+      You will see the one Item you previously created.
+      Add a new Item, and you will see that you can now set the priority.
+      Choose ``High`` for the priority and save the Item.
```
Comment (mongodben) [inline source/tutorial/flutter.txt:221]: note: no copy changes here. just moved text to a new paragraph per above changes.

File: source/tutorial/flutter.txt [hunk 3]
```diff
@@ -209,24 +209,16 @@ Modify the Application
...
-      running the app. Refer to your device or simulator's documentation
-      for help deleting app data.
-
    .. step:: Run and Test
 
-      At this point, you can run the application again. Log in using the account
-      you created earlier in this tutorial. You will see the one Item you
-      previously created. Add a new Item, and you will see that you can now
-      set the priority. Choose ``High`` for the priority and save the Item.
+      Before you run the application again, perform a `hot restart <https://docs.flutter.dev/development/tools/hot-reload#special-cases>`__.
```
Comment (dacharyc) [inline source/tutorial/flutter.txt:214]: The docs you've linked to here refer to this as a "hot reload." Should we do the same for consistency? Or are these two terms used interchangeably in educational material?

**Review by dacharyc — APPROVED** [review]
Overall LGTM - just one Q about the term we're using in a link. I did not, however, actually run through and test this. Do you need me to test, or is it just a copy/formatting review?

File: source/tutorial/flutter.txt (hunk 3 above)
Comment (mongodben) [reply source/tutorial/flutter.txt:214]: no this is a separate thing. tho unfortunately there aren't any good docs on it in the official flutter documentation. 

in that section linked here "special cases" it talks about why you might need to do a hot **restart** (as opposed to the more normal hot **reload**)

**Pull request merged**